federation/pkg/dnsprovider/providers/aws/route53/rrsets.go

```diff
@@ -49,18 +51,31 @@ func (rrsets ResourceRecordSets) List() ([]dnsprovider.ResourceRecordSet, error)
 }
 
 func (rrsets ResourceRecordSets) Get(name string) (dnsprovider.ResourceRecordSet, error) {
-	var newRrset dnsprovider.ResourceRecordSet
-	rrsetList, err := rrsets.List()
+	input := route53.ListResourceRecordSetsInput{
+		HostedZoneId:    rrsets.zone.impl.Id,
+		StartRecordName: aws.String(name),
+		MaxItems:        aws.String("2"), // get 2 items to make sure there is no duplicate records with same DNS name
+	}
+
+	var list []dnsprovider.ResourceRecordSet
+	err := rrsets.zone.zones.interface_.service.ListResourceRecordSetsPages(&input, func(page *route53.ListResourceRecordSetsOutput, lastPage bool) bool {
+		for _, rrset := range page.ResourceRecordSets {
+			if strings.TrimSuffix(name, ".") == strings.TrimSuffix(aws.StringValue(rrset.Name), ".") {
+				list = append(list, &ResourceRecordSet{rrset, &rrsets})
+			}
+		}
+		return true
+	})
 	if err != nil {
 		return nil, err
 	}
-	for _, rrset := range rrsetList {
-		if rrset.Name() == name {
-			newRrset = rrset
-			break
-		}
+	if len(list) == 0 {
+		return nil, nil
+	}
+	if len(list) > 1 {
+		return nil, fmt.Errorf("DNS name %s is ambiguous.", name)
```

Comment:
What I'm thinking is if the DNS record is managed by this code, then it should never have two records with the same name. If it has two records with the same name, then that record has been modified by the user or some other application. I think it's better to log that as an error instead of silently modify the record.